apps/studio/src/server/modules/auth/email/email.router.ts

```diff
@@ -79,6 +86,21 @@ export const emailSessionRouter = router({
     .input(emailVerifyOtpSchema)
     .meta({ rateLimitOptions: {} })
     .mutation(async ({ ctx, input: { email, token } }) => {
+      const oldVerificationToken = await ctx.prisma.verificationToken.findFirst(
+        {
+          where: {
+            identifier: getOtpFingerPrint(email, ctx.req),
+          },
+        },
+      )
+
+      if (!oldVerificationToken) {
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Invalid login email",
+        })
+      }
```

Comment:
[nit] this might be abit confusing to our users because the fingerprint is `email | ip`! hence, this case only happens when their ip chnages. wdyt about "Please request for another OTP"

Reply:
Ah I forgot about the IP part, updated in 6df2f491137cb7321046a905680739bb66f3f381.